src/Data/List/NonEmpty/Base.agda

```diff
@@ -85,6 +85,12 @@ map f (x ∷ xs) = (f x ∷ List.map f xs)
 replicate : ∀ n → n ≢ 0 → A → List⁺ A
 replicate n n≢0 a = a ∷ List.replicate (pred n) a
 
+-- when droping more than the size of the length of the list, the last element remains
+drop : ℕ → List⁺ A → List⁺ A
```

Comment:
This signature feels wrong. Why would you keep the last element? I would imagine that it would return a list rather than a non-empty list?

Can you describe your motivating example for this?

Reply:
the point is one shouldn't drop even the length of the list. essentially the second case shouldn't get hit. the same idea applies for the drop function in List; one shouldn't drop too many, so that an undesired base case is never hit. if drop returns a list, then it is rather useless; as one should have been working with a list to begin with. and even if it returned a list, it's going to be empty if dropped too much, so the problem is really not fixed. In my case, List+ is used for a context, and nonemptyness is part of the preconditions. dropping it should return another genuine context, which is at least nonempty.

Reply:
consider the counterpart in List for a reference:
https://github.com/agda/agda-stdlib/blob/e3533e8ccbef6ebc1558f79f12661f482e57d49e/src/Data/List/Base.agda#L271-L274

Reply:
Call it `drop-all-but-one`? [I don't like that name, but it might spur better suggestions]

Reply:
sounds like that would drop 1 element!  `drop-n-1` ? The second minus should probably be the one with a dot over it. That's exactly what it does...

Reply:
it's not. it's dropping really n. it's just that if you drop too many, it must keep the last element to make the type happy.

Reply:
Indeed, you're right. I don't know, I think I'd prefer `drop` to take a `n < length l` argument. Then have a kind of 'wild' drop that is made artificially total (i.e. your current implementation).

Reply:
this is what pauses me. `drop` in list is the wild version, so `drop` in nonempty list should also be wild. in fact, typesafe versions don't exist in list.

Reply:
I have to confess I don't understand what the problem is with the existing implementation of `drop` in `Data.List.Base`, I think the semantics of returning the empty list when you drop more than the number of elements in the list is perfectly reasonable. I also think I stand by my feeling that the standard operation `drop` should return a list rather than a non-empty list here. So I'm totally on board with @JacquesCarette's suggestion that we find an alternative name for this variant.

Ideally my first suggestion would be to call it `drop⁺`, but we already use `⁺` to denote introduction forms in predicates and relations over data types, so eventually we'd end up with names like `drop⁺⁺`. Wondering if that's a problem... Alternatively we could go for `drop+`?

Reply:
I am ok with any of these.